presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -113,6 +115,9 @@
 
         timeZoneId.set(TimeZone.getDefault().getID());
         locale.set(Locale.getDefault());
+
+        this.queryInterceptorInstances = uri.getQueryInterceptors();
```

Comment:
Let's do a defensive copy just to be safe.
```suggestion
        this.queryInterceptorInstances = ImmutableList.of(uri.getQueryInterceptors());
```

Reply:
Unless I'm missing something, it looks like we can't easily cast a LinkedList to an ImmutableList. Would it make more sense to make the return type of ClassListConverter in AbstractConnectionProperty an ImmutableList instead of a LinkedList? We don't expect to modify it after it's built and I don't think we're seeing many benefits to using a LinkedList for our usage.

Reply:
My suggestion had a typo, it should be:

```java
this.queryInterceptorInstances = ImmutableList.copyOf(uri.getQueryInterceptors());
```